osu.Game/Lists/IntervalList.cs

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using osu.Framework.Lists;
+
+namespace osu.Game.Lists
+{
+    /// <summary>
+    /// Represents a list of intervals that can be used for whether a specific value falls into one of them.
+    /// </summary>
+    /// <typeparam name="T">The type of interval values.</typeparam>
+    public class IntervalList<T> : ICollection<Interval<T>>, IReadOnlyList<Interval<T>>
```

Comment:
this doesn't make sense for things that are not numbers (there is no such thing as an interval of strings). if C# had traits I'd use that, but it doesn't so why not just use `float` or something?